### PR TITLE
Profiler Extension

### PR DESCRIPTION
**Pull request by nehlsen**
This PR drops a 🔥 extension for the Symfony Profiler. Now you can peep 👀 all the loaded fixtures, see how long each one took to load ⏱️, and check out the memory each one gobbled up 🧠. Straight fire! 🚀

### CODE REVIEW

**Review by lukadschaak — APPROVED** [review]

**Review by lukadschaak — CHANGES_REQUESTED** [review]
Pls fix phpstan

**Comment by nehlsen** [comment]
@lukadschaak done! :boom:

**Review by lukadschaak — APPROVED** [review]

File: src/Profiler/DataCollector/FixtureLoaderCollector.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+use Neusta\Pimcore\FixtureBundle\Profiler\FixtureReference\FixtureReference;
+use Neusta\Pimcore\FixtureBundle\Profiler\FixtureReference\FixtureReferenceResolver;
+use Neusta\Pimcore\FixtureBundle\Profiler\PerformanceInfo\PerformanceInfo;
+use Neusta\Pimcore\FixtureBundle\Profiler\Timing\TimingCollector;
+use Symfony\Bundle\FrameworkBundle\DataCollector\AbstractDataCollector;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class FixtureLoaderCollector extends AbstractDataCollector
```
Comment (jdreesen) [inline src/Profiler/DataCollector/FixtureLoaderCollector.php:13]: What do you think about making everything `final` and `@internal`? This is no public API, right?

```suggestion
/**
 * @internal
 */
final class FixtureLoaderCollector extends AbstractDataCollector
```

File: src/Profiler/EventListener/FixtureDependencyGraphSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+namespace Neusta\Pimcore\FixtureBundle\Profiler\EventListener;
+
+use Neusta\Pimcore\FixtureBundle\Event\AfterLoadFixtures;
+use Neusta\Pimcore\FixtureBundle\Event\BeforeExecuteFixture;
+use Neusta\Pimcore\FixtureBundle\Fixture\Fixture;
+use Neusta\Pimcore\FixtureBundle\Profiler\FixtureReference\FixtureReferenceResolver;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class FixtureDependencyGraphSubscriber implements EventSubscriberInterface
```
Comment (jdreesen) [inline src/Profiler/EventListener/FixtureDependencyGraphSubscriber.php:11]:
```suggestion
/**
 * @internal
 */
final class FixtureDependencyGraphSubscriber implements EventSubscriberInterface
```

File: src/Profiler/EventListener/TimingSubscriber.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+namespace Neusta\Pimcore\FixtureBundle\Profiler\EventListener;
+
+use Neusta\Pimcore\FixtureBundle\Event\AfterExecuteFixture;
+use Neusta\Pimcore\FixtureBundle\Event\BeforeExecuteFixture;
+use Neusta\Pimcore\FixtureBundle\Profiler\Timing\Timing;
+use Neusta\Pimcore\FixtureBundle\Profiler\Timing\TimingCollector;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class TimingSubscriber implements EventSubscriberInterface
```
Comment (jdreesen) [inline src/Profiler/EventListener/TimingSubscriber.php:11]:
```suggestion
/**
 * @internal
 */
final class TimingSubscriber implements EventSubscriberInterface
```

File: src/Profiler/FixtureReference/FixtureReference.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
+<?php declare(strict_types=1);
+
+namespace Neusta\Pimcore\FixtureBundle\Profiler\FixtureReference;
+
+use Neusta\Pimcore\FixtureBundle\Fixture\Fixture;
+
+class FixtureReference
```
Comment (jdreesen) [inline src/Profiler/FixtureReference/FixtureReference.php:7]:
```suggestion
/**
 * @internal
 */
final class FixtureReference
```

File: src/Profiler/FixtureReference/FixtureReferenceResolver.php [hunk 5]
```diff
@@ -0,0 +1,103 @@
+<?php declare(strict_types=1);
+
+namespace Neusta\Pimcore\FixtureBundle\Profiler\FixtureReference;
+
+use Neusta\Pimcore\FixtureBundle\Fixture\Fixture;
+use Neusta\Pimcore\FixtureBundle\Fixture\HasDependencies;
+
+class FixtureReferenceResolver
```
Comment (jdreesen) [inline src/Profiler/FixtureReference/FixtureReferenceResolver.php:8]:
```suggestion
/**
 * @internal
 */
final class FixtureReferenceResolver
```

File: src/Profiler/PerformanceInfo/PerformanceInfo.php [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<?php declare(strict_types=1);
+
+namespace Neusta\Pimcore\FixtureBundle\Profiler\PerformanceInfo;
+
+class PerformanceInfo
```
Comment (jdreesen) [inline src/Profiler/PerformanceInfo/PerformanceInfo.php:5]:
```suggestion
/**
 * @internal
 */
final class PerformanceInfo
```

File: src/Profiler/Timing/Timing.php [hunk 7]
```diff
@@ -0,0 +1,53 @@
+<?php declare(strict_types=1);
+
+namespace Neusta\Pimcore\FixtureBundle\Profiler\Timing;
+
+use Neusta\Pimcore\FixtureBundle\Fixture\Fixture;
+use Neusta\Pimcore\FixtureBundle\Profiler\PerformanceInfo\PerformanceInfo;
+use Symfony\Component\Stopwatch\Stopwatch;
+
+class Timing
```
Comment (jdreesen) [inline src/Profiler/Timing/Timing.php:9]:
```suggestion
/**
 * @internal
 */
final class Timing
```

File: src/Profiler/Timing/TimingCollector.php [hunk 8]
```diff
@@ -0,0 +1,25 @@
+<?php declare(strict_types=1);
+
+namespace Neusta\Pimcore\FixtureBundle\Profiler\Timing;
+
+use Neusta\Pimcore\FixtureBundle\Fixture\Fixture;
+use Neusta\Pimcore\FixtureBundle\Profiler\PerformanceInfo\PerformanceInfo;
+
+class TimingCollector
```
Comment (jdreesen) [inline src/Profiler/Timing/TimingCollector.php:8]:
```suggestion
/**
 * @internal
 */
final class TimingCollector
```

File: config/services.yaml [hunk 9]
```diff
@@ -31,15 +39,10 @@ services:
...
-    arguments:
-      $fixtureLocator: '@Neusta\Pimcore\FixtureBundle\Locator\AllFixturesLocator'
-
-  Neusta\Pimcore\FixtureBundle\FixtureLoader\SelectiveFixtureLoader:
-    arguments:
-      $fixtureLocator: '@Neusta\Pimcore\FixtureBundle\Locator\NamedFixtureLocator'
     public: true
 
+  Neusta\Pimcore\FixtureBundle\Profiler\:
+    resource: '../src/Profiler/*'
```
Comment (jdreesen) [inline config/services.yaml:45]: I think this should only be loaded in `debug` mode, because the profiler is a development tool.

**Review by jdreesen — APPROVED** [review]
This looks great 😍 Thank you very much!

Just some nits.

**Review by jdreesen — APPROVED** [review]

**Pull request merged**